app/src/main/java/com/example/android_level_3/FragmentAddContact.kt

```diff
@@ -0,0 +1,219 @@
+package com.example.android_level_3
+
+import android.os.Bundle
+import android.text.Editable
+import android.text.TextWatcher
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.activityViewModels
+import androidx.lifecycle.MutableLiveData
+import androidx.navigation.fragment.findNavController
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.android_level_3.adapter.ContactAdapter
+import com.example.android_level_3.adapter.ElementClickListener
+import com.example.android_level_3.databinding.FragmentAddContactBinding
+import com.example.android_level_3.retrofit.model.Contact
+import com.example.android_level_3.viewmodel.MainViewModel
+import com.google.android.material.snackbar.Snackbar
+
+class FragmentAddContact : Fragment(), ElementClickListener {
+
+    private lateinit var binding: FragmentAddContactBinding
+    private lateinit var recyclerViewAdapter: ContactAdapter
+    private val viewModel: MainViewModel by activityViewModels()
+
+    private val isVisibleProgressBar = MutableLiveData(false)
+    private var connectionErrorSnackbar: Snackbar? = null
+    private var contactId: Int? = null
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        binding = FragmentAddContactBinding.inflate(inflater, container, false)
+
+        setObservers()
+        setListeners()
+        requestGetAllUsers()
+
+        return binding.root
+    }
+
+    private fun setObservers() {
+
+        viewModel.listOfAllUsers.observe(viewLifecycleOwner) {                         // отслеживание и отображение списка пользователей
+            if (viewModel.isActiveSearchAddContact.value == true) {
```

Comment:
if the value is a Boolean, you can keep it simpler with
 if (viewModel.isActiveSearchAddContact.value)

Reply:
> if the value is a Boolean, you can keep it simpler with if (viewModel.isActiveSearchAddContact.value)

unless it's nullable of course, which is probably the case due to a live data

Reply:
> > if the value is a Boolean, you can keep it simpler with if (viewModel.isActiveSearchAddContact.value)
> 
> unless it's nullable of course, which is probably the case due to a live data

true, my bad